README.md

```diff
@@ -23,6 +35,72 @@ The element requires a sample configuration with one property like the one below
 }
 ```
 
+## Structure of the Kontent.ai Custom Element
+
+### Link the Custom Element API
+
+Every Kontent.ai custom element needs the [Custom Element API](https://kontent.ai/learn/reference/custom-elements-js-api/) to work properly. You should include it in your `html` file like this: https://github.com/kontent-ai/custom-element-template-react/blob/e184179039aa705a82722d778e503dfb511f7115/public/index.html#L8-L10
+
+#### (Optionally) Include Kontent.ai styles
+
+If you want your custom element to look like a part of the Kontent.ai app, you'll need to include the Kontent.ai CSS file. You can find it [here in this repository](https://github.com/kontent-ai/custom-element-template-react/blob/main/public/kontent-ai-app-styles.css). You have to include the [font file](https://github.com/kontent-ai/custom-element-template-react/blob/main/public/kontent-ai-icons-v3.0.1.woff2) as well. https://github.com/kontent-ai/custom-element-template-react/blob/e184179039aa705a82722d778e503dfb511f7115/public/index.html#L5-L7
+
+### Initialize the custom element
+
+Before you start any interaction with the Kontent.ai app, you'll need **initialize the custom element**. You can do that by calling the `init` function from the provided API like so https://github.com/kontent-ai/custom-element-template-react/blob/e184179039aa705a82722d778e503dfb511f7115/src/IntegrationApp.tsx#L18-L29
+and waiting until the callback you passed into the `init` function is called. You'll receive the custom element configuration object of your specification, as well as additional context - more on that in [the documentation](https://kontent.ai/learn/reference/custom-elements-js-api/#a-init-method). 
+You can leverage state in React to store the important configuration while setting it inside of the initialization callback. 
+
+```ts
+const [config, setConfig] = useState<Config | null>(null);
+
+useEffect(() => {
+  CustomElement.init((element, context) => {
+    if (!isValidConfig(element.config)) {
+      throw new Error('Not the config this element expects');
+    }
+    setConfig(element.config);
+    // more logic
+  });
+}, []);
+```
+
+### Disable your element if needed
+
+In certain circumstances, the Kontent.ai app might instruct your custom element to display itself in a **disabled state**. Disabled state means that the value of your element cannot be changed. This can happen when the edited item is published (therefore cannot be edited), or the current user does not have permission to edit the custom element.
```

Comment:
the current user does not have permission to edit the custom element. -> this information is not in Custom element API - or is it?

Reply:
You are correct, it is -> https://kontent.ai/learn/reference/custom-elements-js-api/#a-ondisabledchanged-method